source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -98,15 +101,28 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool end_st
   return status;
 }
 
-Http::FilterDataStatus Filter::onData(ProcessorState& state, ProcessingMode::BodySendMode body_mode,
-                                      Buffer::Instance& data, bool end_stream) {
-  if (state.callbackState() == ProcessorState::CallbackState::Headers) {
+Http::FilterDataStatus Filter::onData(ProcessorState& state, Buffer::Instance& data,
+                                      bool end_stream) {
+  if (end_stream) {
+    state.setCompleteBodyDelivered(true);
```

Comment:
Non-actionable comment but I'm interested in your take; how adversarial are we treating remote ext_proc servers as being? One use case that might be interesting for `ext_proc` is allowing service chaining where trust relationships between infrastructure providers and tenants imply a threat model that doesn't fully trust the `ext_proc` server. Any thoughts on what would be required to make this fully robust?

Reply:
Right now the relationship is pretty trusting -- ext_proc servers can set most headers, and they can manipulate the body and trailers too. There is an "ignore errors" setting which makes Envoy more resilient to a server failure, but without it a flaky server can definitely cause errors in the proxy as well. 

I think that if we don't trust servers, then the user should certainly set up ext_proc to ignore errors from servers, and we might want to consider a model in which a server is limited by the proxy as to what it is allowed to change in the headers, trailers, and body.